src/muz/base/rule_properties.cpp

```diff
@@ -256,18 +258,62 @@ bool rule_properties::check_accessor(app* n) {
         if (!use_list.contains(e))
             return false;
         for (expr* parent : use_list[e]) {
-            if (!parent)
-                return false; // top-level expressions are not guarded
-            if (is_recognizer(parent)) 
-                continue;
-            if (m.is_ite(parent) && to_app(parent)->get_arg(1) == e && is_recognizer(to_app(parent)->get_arg(0))) 
+            if (!parent) { // top-level expression
+                // check if n is an unguarded "else" branch
+                ptr_vector<func_decl> ctors;
+                for (auto *rc : recognizers)
+                    if (to_app(rc)->get_arg(0) == n->get_arg(0))
+                        ctors.push_back(m_dt.get_recognizer_constructor(to_app(rc)->get_decl()));
+                ptr_vector<func_decl> diff;
+                for (auto *dtc : *m_dt.get_datatype_constructors(s))
+                    if (std::find(ctors.begin(), ctors.end(), dtc) == ctors.end())
```

Comment:
if (!ctx.contains(dtx))

Reply:
Shorter:

```
obj_hashtable<func_decl> ctors;
for (auto *rc : recognizers)
        if (to_app(rc)->get_arg(0) == n->get_arg(0))
             ctors.insert(m_dt.get_recognizer_constructor(to_app(rc)->get_decl()));
if (m_dt.get_datatype_constructors(s)->size() != ctors.size() + 1)
    return false;
continue;
```

Reply:
Also, seems instead of first collecting recognizers, then add them to ctors, just collect ctors directly.

auto add_not_recognizer = [&](expr* r) {
    if (!m.is_not(r, r))
        return;
    if (!m_dt.is_recognizer(r))
        return;
    if (n->get_arg(0) != to_app(r)->get_arg(0))
        return;
     ctors.insert(m_dt.get_recognizer_constructor(to_app(r)->get_decl()); 
};